test/cocotb/osdtestlib/modtestclasses.py

```diff
@@ -0,0 +1,637 @@
+                                   destination = destination, source = source, 
+                                   packet_type = self.PACKET_TYPE_REG, 
+                                   packet_subtype = \
+                                   self.PACKET_SUBTYPE_REQ_WRITE_REG_16, 
+                                   message = content)
+        elif reg_width == 32:
+            # send a write request to the dut
+            yield self.send_packet(dut = dut, packet_driver = write_driver, 
+                                   packet_utility = write_utility, 
+                                   destination = destination, source = source, 
+                                   packet_type = self.PACKET_TYPE_REG, 
+                                   packet_subtype = \
+                                   self.PACKET_SUBTYPE_REQ_WRITE_REG_32, 
+                                   message = content)
+        elif reg_width == 64:
+            # send a write request to the dut
+            yield self.send_packet(dut = dut, packet_driver = write_driver, 
+                                   packet_utility = write_utility, 
+                                   destination = destination, source = source, 
+                                   packet_type = self.PACKET_TYPE_REG, 
+                                   packet_subtype = \
+                                   self.PACKET_SUBTYPE_REQ_WRITE_REG_64, 
+                                   message = content)
+        elif reg_width == 128:
+            # send a write request to the dut
+            yield self.send_packet(dut = dut, packet_driver = write_driver, 
+                                   packet_utility = write_utility, 
+                                   destination = destination, source = source, 
+                                   packet_type = self.PACKET_TYPE_REG, 
+                                   packet_subtype = \
+                                   self.PACKET_SUBTYPE_REQ_WRITE_REG_128, 
+                                   message = content)
+        else:
+            # tell the user that an unsupported register width was chosen
+            raise TestFailure("Invalid register width parameter")
+        
+        # read in the response of the debug module
+        response = []
+        yield self.receive_packet(dut = dut, response = response)
+        
+        # if the module returned a success, everything is fine
+        if ((response[2] & 0x3C00) >> 10) == \
+        self.PACKET_SUBTYPE_REQ_WRITE_REG_SUCCESS:
+            dut._log.info("Write process was successful!")
+        # if it returned an error, output that error
+        elif ((response[2] & 0x3C00) >> 10) == \
+        self.PACKET_SUBTYPE_REQ_WRITE_REG_ERROR:
+            dut._log.info("An error occurred during the write process!")
+        # otherwise something else went wrong
+        else:
+            dut._log.info("An unidentified error occurred!")
+                
+class Utility:
```

Comment:
Please use a more descriptive name. What kind of utility functions does this class contain? (Also, most likely all methods of this class will go away with a new class representing a packet.)